### DISCUSSION TITLE
Boot fails after flashing esp32s3

### DISCUSSION THREAD

**Discussion by Leo5126** [micropython/micropython#10187] [ESP32] [answered]
Hello,

I'm currently trying to flash ESP32-S3-DevKitC-1-N8R2 with this firmware (v1.19.bin)  https://micropython.org/download/GENERIC_S3_SPIRAM/
Flashing works fine, but when i connect through the serial monitor i get this output:

```
Rebooting...
�ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x420daf32
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0xf24
load:0x403b6000,len:0xc2c
load:0x403ba000,len:0x2d60
entry 0x403b61f4
E (224) psram: PSRAM ID read error: 0x00ffffff
E (224) spiram: SPI RAM enabled but initialization failed. Bailing out.
E (10) spiram: SPI RAM not initialized
```

I've flashed with this command:

```
python -m esptool --chip esp32s3 --port COM11 --before default_reset --after hard_reset --chip esp32s3 write_flash --flash_mode dio --flash_size 2MB --flash_freq 40m -z 0 C:\_Firmware\GENERIC_S3_SPIRAM-20220618-v1.19.1.bin
```
I've tried also other settings for write_flash (flash_mode, flash_size, flash_freq

When I flash the firmware that hasn't SPIRAM included, all works fine after boot.
_Edit: The other Firmwareversion is now also crashing on boot._ 
```
rst:0x3 (RTC_SW_SYS_RST),boot:0x18 (SPI_FAST_FLASH_BOOT)
Saved PC:0x403bb412
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0xf24
load:0x403b6000,len:0xc2c
load:0x403ba000,len:0x2d60
entry 0x403b61f4
E (29) flash_parts: partition 0 invalid magic number 0xaa
E (29) boot: Failed to verify partition table
E (29) boot: load partition table error!
```

So now i have two questions:
- Does this board even support SPIRAM?
- If yes, is there a way to fix the issue i get in the serial monitor?

Thanks alot in advance for your help.

**Comment by ThinkTransit** [comment]
I have been using the S2 and S3 modules for a while and haven't seen that one before.

Do you have another module you can test on?

**Reply by Leo5126** [reply]
> Do you have another module you can test on?

Before i've been using an ESP32-WROOM-32D and there I haven't got any problems. But that model is dicontinued so I'm testing a new one.

I've tried now two different hardware modules (ESP32-WROOM-1-N8, ESP32-WROOM-1U-N8R8). At that point i'm getting on both systems an error, when i connect the serial monitor, that looks quite similar nomatter if i'm flashing the SPIRAM Firmware or the normal one. 
The interesting thing is, that there isn't really any error message as i got before and mentioned before.

I've flashed again both versions and I get this response in the Serial Monitor:
```
invalid header: 0xcd4b3048
```

My guess was before that the settings in the firmware might not quite match the hardware. But what I can't explain is that with both versions I get the same error. 

To rule out the possibility that it is the firmware, I am currently trying to create my own firmware build with the specific settings of my hardware. Unfortunately, I have had no success so far, because I have encountered some problems. I think an update of the docs is necessary there. Further it would be nice if there would be an update about flashing the firmware. I can find alot of different information about it, but nothing that really guides through the whole process.

**Reply by ThinkTransit** [reply]
Have you tried the instructions here

https://github.com/micropython/micropython/tree/master/ports/esp32

**Reply by Leo5126** [reply]
Yes, this is what I'm trying to do currently. But I'm stuck at the point https://github.com/micropython/micropython/tree/master/ports/esp32#building-the-firmware . When I try to build "mpy-cross" i get this error, which I currently try to fix:

```
process_begin: CreateProcess(NULL, uname -s, ...) failed.
process_begin: CreateProcess(NULL, gcc -dumpmachine, ...) failed.
process_begin: CreateProcess(NULL, gcc -dumpmachine, ...) failed.
make: Entering directory 'C:/_project/17_firmwareBuild/mpy_repo/micropython/mpy-cross'
mkdir -p build/genhdr
Syntaxfehler.
../py/mkrules.mk:150: recipe for target 'build/genhdr' failed
make: *** [build/genhdr] Error 1
make: Leaving directory 'C:/_project/17_firmwareBuild/mpy_repo/micropython/mpy-cross'
```

**Reply by jimmo** [reply]
It looks like maybe the build environment isn't set up (e.g. gcc, etc). Maybe try using Docker instead? See in-progress instructions in this PR: https://github.com/micropython/micropython/pull/9996

**Reply by Leo5126** [reply]
hi @jimmo,
Thanks for the tipp. I've got it working with the WSL application, I have constantly encountered further errors so I have not tried further via windows itself. I didn't tried the doker version until now.

**Comment by jimmo** [comment, marked as answer]
> I've tried now two different hardware modules (ESP32-WROOM-1-N8, ESP32-WROOM-1U-N8R8). 

The N8 doesn't have SPIRAM. The N8R8 does, but it's Octal spiram, which isn't supported yet -- see https://github.com/micropython/micropython/pull/8689 it will be added soon!

**Reply by Leo5126** [reply]
Ah ok, meanwhile i got it running with my own firmware build. But i still do not know why even the standard generic-s3 version did not work. At least now i know that SPIRAM is not working at the moment ;)

**Comment by Hamo-creator** [comment]
Hello Leo5126,
i am using the same Board ESP32-S3-DevKitC-1-N8R2 but i am not been able till now to flash the code on it.
I am using micropython and my ide is Thonny.
I installed the same bin File GENERIC_S3.
I installed esptool too.
That was all what i could manage.
Could you help me with that?
How could you manage that?
Thanks

**Reply by Leo5126** [reply]
Hi @Hamo-creator,

I've solved the problem by creating my own firmware. Have a look [here](https://github.com/micropython/micropython/tree/master/ports/esp32) how to do that.  When doing the build I recommend you doing that either on a ubuntu or on the Windows Subsystem for Linux. 

Unfortunately I cant find the build that worked for me. But I've written me a short guide about the build process, maybe this can help you additionally to the guide that is given on the site i've linked you. I've attached you the guide.

[flashing.md](https://github.com/micropython/micropython/files/10608327/flashing.md)

**Reply by Hamo-creator** [reply]
Thank you.
That helped me a lot.
